src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperUsageDescriptorFactory.cs

```diff
@@ -0,0 +1,141 @@
+                assemblyLocation = Uri.UnescapeDataString(uri.Path);
+
+                // Still couldn't resolve a valid assemblyLocation.
+                if (string.IsNullOrWhiteSpace(assemblyLocation))
+                {
+                    return null;
+                }
+            }
+
+            var xmlDocumentationFile = GetXmlDocumentationFile(assemblyLocation);
+
+            // We only want to process the file if it exists.
+            if (xmlDocumentationFile != null)
+            {
+                var documentationProvider = new XmlDocumentationProvider(xmlDocumentationFile.FullName);
+
+                var summary = documentationProvider.GetSummary(id);
+                var remarks = documentationProvider.GetRemarks(id);
+
+                if (!string.IsNullOrEmpty(summary) || !string.IsNullOrEmpty(remarks))
+                {
+                    return new TagHelperUsageDescriptor(summary, remarks);
+                }
+            }
+
+            return null;
+        }
+
+        private static FileInfo GetXmlDocumentationFile(string assemblyLocation)
+        {
+            if (string.IsNullOrWhiteSpace(assemblyLocation) ||
+                assemblyLocation.IndexOfAny(InvalidPathCharacters) != -1)
+            {
+                return null;
+            }
+
+            try
+            {
+                var assemblyDirectory = Path.GetDirectoryName(assemblyLocation);
+                var assemblyName = Path.GetFileName(assemblyLocation);
+                var assemblyXmlDocumentationName = Path.ChangeExtension(assemblyName, ".xml");
+                var culture = CultureInfo.CurrentCulture;
+
+                var assemblyXmlDocumentationFile = new FileInfo(
+                    Path.Combine(assemblyDirectory, assemblyXmlDocumentationName));
+
+                // If there's not an XML file side-by-side the .dll it may exist in a culture specific directory.
+                if (!assemblyXmlDocumentationFile.Exists)
+                {
+                    var fallbackDirectories = GetCultureFallbackDirectories();
+                    assemblyXmlDocumentationFile = fallbackDirectories
+                        .Select(fallbackDiretory =>
+                            new FileInfo(
+                                Path.Combine(assemblyDirectory, fallbackDiretory, assemblyXmlDocumentationName)))
+                        .FirstOrDefault(file => file.Exists);
+                }
+
+                return assemblyXmlDocumentationFile;
+            }
+            catch (PathTooLongException)
+            {
+                // Could not resolve XML file.
+                return null;
+            }
+        }
+
+        private static IEnumerable<string> GetCultureFallbackDirectories()
+        {
+            var culture = CultureInfo.CurrentCulture;
+
+            // Following the fall-back process defined by:
+            // https://msdn.microsoft.com/en-us/library/sb6a8618.aspx#cpconpackagingdeployingresourcesanchor1
```

Comment:
@Eilon talked with @ToddGrun about how the refs are picked up to provide localized text for other pieces in the editor (HTML editor etc.) and he said that it's just handled via `CultureInfo.CurrentCulture`, they don't do anything special to consume the VS specific setting.

@dougbu This isn't for DNX created packages: today DNX does not support satellite assemblies; @davidfowl said it's currently on the todo. When they do light up support for localized bits we'll revisit this and add support if it's needed (hopefully just works).